libcontainerd/client_linux.go

```diff
@@ -563,6 +563,8 @@ func (clnt *client) Restore(containerID string, options ...CreateOption) error {
 		clnt.remote.Lock()
 		return nil
 	}
+	// relock because of the defer
+	clnt.remote.Lock()
```

Comment:
Now that I think about it, what about removing all those extra `clnt.remote.Lock()` and adding a `defer clnt.remote.Unlock()` on line 549?